### PR TITLE
FAQ Section: Let's Encrypt

### PR DESCRIPTION
**Pull request by enckse**
how do you do let's encrypt? let's move this from an issue to the FAQ, this probably takes care of #118 and #231 well enough to at least resolve them and start using the md/docs going forward

### CODE REVIEW

**Comment by DanielOaks** [comment]
nice. yeah I'm pretty happy with this as written here, thanks a bunch for contributing this change!

**Comment by slingamn** [comment]
With these instructions as written, I think people are likely to encounter a permissions issue (by default, non-root cannot read certbot's certificates).

**Comment by enckse** [comment]
I don't disagree that there may be other issues that are encountered but in my writing's defense:
* I've only production setup (with LE) one instance of oragono and that's in a container so it does have different setup requirements/complexities
* It's a better starting point than arbitrary issues without a single point to start from

If we have a starting point somewhere I'd hope we can make it better over time :/

**Comment by slingamn** [comment]
Sorry, just to clarify, I think this change is great! I think we should just call out the permissions issue because people are quite likely to be surprised by it (I got a question in IRC about it last week) and are also likely to be tempted to resolve it by running oragono as root (which is bad).

We could just mention something about the permissions issue and suggest that they use the [renewal-hooks](https://certbot.eff.org/docs/using.html) feature of certbot to work around it. Unfortunately there is not a standard way to install the certificates for the use of an unprivileged user. This is what we hacked together:

https://github.com/darwin-network/slash/blob/master/etc/bin/install-lecerts

**Comment by slingamn** [comment]
(renewal-hooks may also be an improvement on modifying Certbot's systemd unit file)

**Comment by enckse** [comment]
Totally understand, trying (maybe not successfully) to say "I can't write what I'm not doing." because we'll definitely have me shooting from the hip on how to do stuff!

I 100% agree that people will have permissions issues. I also, while writing, was like "how much 'help' should I give when dealing with let's encrypt/certbot" and immediately that's not oragono's problem is the decision I made. I've pushed a small "caveats" update that I hope starts us down the path of known issues that maybe we can't fix for each user or just like "Yeah, that sucks be ready". My own deployment is containerized so I have my own injection points but not everybody is going to do that either :/ This probably gets worse when we start to talk about how various distributions may or may not prefer to solve problems like this. None of which is probabliy a good discussion for this PR but I definitely know this whole topic is a mess so whatever we can do to lessen the confusion the better!

File: docs/MANUAL.md [hunk 1]
```diff
@@ -225,6 +225,9 @@ Otherwise, in the Oragono config file, you'll want to enable raw line logging by
 4. If you are using auto-renew via Let's Encrypt you may want to have a service or timer send a SIGHUP to the oragono process to reload the configuration and certs
     1. e.g. you could edit the `certbot.service` and add the following `ExecStartPost=/usr/bin/kill -HUP $(/usr/bin/pidof oragono)`
 
+Caveats:
+* Depending on how and who you run oragono as, you may run into permissions issues with the certificates as, by default, Let's Encrypt will generate certificates non-root users cannot read
```
Comment (DanielOaks) [inline docs/MANUAL.md]: Just curious, can we add this to the end of the line?

```
To alleviate this, there's an example bash script [linked here](https://github.com/darwin-network/slash/blob/master/etc/bin/install-lecerts) that may be useful
```

I'm fine linking to that and, again as mentioned, improving it further as we go

**Comment by DanielOaks** [comment]
yeah I can see watcha mean about it not being ora's problem to solve for sure. I tend to err on the side of a bit more assistance from the docs but linking to external resources as much as possible. will merge this, throw that text on top of it and we can go forward with similar issues / contributions in the same way. totally happy to see more contributions if they're done in a similar way to this one :)

**Pull request merged**